airbyte-integrations/connectors/destination-qdrant/destination_qdrant/destination.py

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+import re
+import uuid
+from typing import Any, Iterable, List, Literal, Mapping, Optional, Union
+
+import dpath.util
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.destinations.vector_db_based.batcher import Batcher
+from airbyte_cdk.destinations.vector_db_based.config import (
+    CohereEmbeddingConfigModel,
+    FakeEmbeddingConfigModel,
+    OpenAIEmbeddingConfigModel,
+    ProcessingConfigModel,
+)
+from airbyte_cdk.destinations.vector_db_based.document_processor import (
+    METADATA_RECORD_ID_FIELD,
+    METADATA_STREAM_FIELD,
+    Chunk,
+    DocumentProcessor,
+)
+from airbyte_cdk.destinations.vector_db_based.embedder import CohereEmbedder, Embedder, FakeEmbedder, OpenAIEmbedder
+from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
+from airbyte_cdk.destinations.vector_db_based.utils import format_exception
+from airbyte_cdk.destinations.vector_db_based.writer import Writer
+from airbyte_cdk.models import (
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    ConfiguredAirbyteCatalog,
+    ConnectorSpecification,
+    Status,
+    Type,
+)
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+from jsonschema import RefResolver
+from pydantic import BaseModel, Field
+from qdrant_client import QdrantClient, models
+from qdrant_client.conversions.common_types import PointsSelector
+
+BATCH_SIZE = 256
+
+
+class UsernamePasswordAuth(BaseModel):
+    mode: Literal["username_password"] = Field("username_password", const=True)
+    username: str = Field(..., title="Username", description="Username for the Qdrant instance")
+    password: str = Field(..., title="Password", description="Password for the Qdrant instance", airbyte_secret=True)
```

Comment:
There is no built-in user-password authentication. We support API key headers as the only way.